src/common/modals/JavaSetup.js

```diff
@@ -251,16 +291,42 @@ const AutomaticSetup = () => {
 
     await fse.remove(downloadLocation);
 
+    const firstExtractionJava16 = extractFull(
+      downloadjava16Location,
+      tempFolder,
+      {
+        $bin: sevenZipPath,
+        $progress: true
+      }
+    );
+
+    await new Promise((resolve, reject) => {
+      firstExtractionJava16.on('progress', ({ percent }) => {
+        ipcRenderer.invoke('update-progress-bar', percent);
+        setDownloadPercentage(percent);
+      });
+      firstExtractionJava16.on('end', () => {
+        resolve();
+      });
+      firstExtractionJava16.on('error', err => {
+        reject(err);
+      });
+    });
+
+    await fse.remove(downloadjava16Location);
+
     // If NOT windows then tar.gz instead of zip, so we need to extract 2 times.
     if (process.platform !== 'win32') {
       ipcRenderer.invoke('update-progress-bar', -1);
       setDownloadPercentage(null);
       await new Promise(resolve => setTimeout(resolve, 500));
       setCurrentStep(`Extracting 2 / ${totalSteps}`);
+
```

Comment:
Total steps is not adjusted to be the sum of all operations

Reply:
Also resolved, show above ;)